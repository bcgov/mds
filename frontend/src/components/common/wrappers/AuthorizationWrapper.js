import React from "react";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";
import { getUserAccessData } from "@/selectors/authenticationSelectors";
import { USER_ROLES } from "@/constants/environment";

/**
 * @constant AuthorizationWrapper conditionally renders react children depending
 * if the role passed in matches the user permissions.
 * 
 * NOTE: If the childComponent is an ant design child component**
 * IE. Menu.Item,
 * <Menu>
 *    <Menu.Item />
 * </Menu>
 *  The implicit flow of passing props down to children could be interrupted.
 * Ie, Menu passes onItemHover down to Menu.Item, with AuthorizationWrapper wrapping Menu.Item, it no longer has access
 * to onItemHover, as a solution, a <div className="custom-menu-item" /> will be used in place of Menu.Item and will inherit all CSS from 
 * Menu.Item.
 * NavBar.js use case example: 
 * <Menu>
    <AuthorizationWrapper permission="role_admin">
      <div className="custom-menu-item">
        <Link to={router.ADMIN_DASHBOARD.route}>
          Admin
        </Link>
      </div>
    </AuthorizationWrapper>
  </Menu>
 * 
 * NOTE: isMajorMine comes from `mine.major_mine_ind`, currently in MDS only Major mines can be updated, 
 * therefore all eidt buttons will be hidden from regional Mines.
 */

const propTypes = {
  permission: PropTypes.string.isRequired,
  isMajorMine: PropTypes.bool,
  children: PropTypes.element.isRequired,
};

const defaultProps = {
  isMajorMine: true,
};
export const AuthorizationWrapper = (props) =>
  props.userRoles.includes(USER_ROLES[props.permission]) &&
  props.isMajorMine && <div>{...props.children}</div>;

AuthorizationWrapper.propTypes = propTypes;
AuthorizationWrapper.defaultProps = defaultProps;

const mapStateToProps = (state) => ({
  userRoles: getUserAccessData(state),
});

export default connect(mapStateToProps)(AuthorizationWrapper);
