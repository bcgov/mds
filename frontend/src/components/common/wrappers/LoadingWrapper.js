import React from "react";
import { PropTypes } from "prop-types";
import { Spin, Icon } from "antd";
import { TABLE_SKELETON } from "@/constants/assets";
import * as Style from "@/constants/styles";

/**
 * @constant LoadingWrapper renders react children || or loading view
 */

const propTypes = {
  condition: PropTypes.bool.isRequired,
  children: PropTypes.element.isRequired,
  type: PropTypes.oneOf(["spinner", "table-skeleton"]),
};

const defaultProps = {
  type: "spinner",
};

export const LoadingWrapper = (props) => {
  const antIcon = (
    <Icon type="loading" style={{ fontSize: 80, color: Style.COLOR.mediumGrey }} spin />
  );

  const renderLoading = (
    <div className="page__content">
      {props.type === "table-skeleton" && (
        <img src={TABLE_SKELETON} alt="loading" id="table-skeleton" />
      )}
      {props.type === "spinner" && (
        <div className="loading-screen--small">
          <Spin id="spinner" indicator={antIcon} />
        </div>
      )}
    </div>
  );

  return (
    <div>
      {props.condition ? (
        <div className="fade-in">{props.children}</div>
      ) : (
        <div>{renderLoading}</div>
      )}
    </div>
  );
};

LoadingWrapper.propTypes = propTypes;
LoadingWrapper.defaultProps = defaultProps;

export default LoadingWrapper;
