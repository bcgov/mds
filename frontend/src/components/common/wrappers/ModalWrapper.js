import React, { Component } from "react";
import { bindActionCreators } from "redux";
import LoadingBar from "react-redux-loading-bar";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Modal, Icon, Button, Popconfirm } from "antd";
import * as Styles from "@/constants/styles";
import { closeModal } from "@/actions/modalActions";
import {
  getIsModalOpen,
  getProps,
  getContent,
  getClearOnSubmit,
  getWidthSize,
} from "@/selectors/modalSelectors";

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool.isRequired,
  content: PropTypes.func,
  props: PropTypes.objectOf(PropTypes.string),
  clearOnSubmit: PropTypes.bool.isRequired,
  widthSize: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
};

const defaultProps = {
  props: {
    title: "",
  },
  content: () => {},
};

export class ModalWrapper extends Component {
  constructor(props) {
    super(props);
    // listens for browser back || forward button click and invokes function to close the modal,
    window.onpopstate = this.onBrowserButtonEvent;
  }

  onBrowserButtonEvent = () => {
    this.props.closeModal();
  };

  renderTitle = () => (
    <div className="inline-flex between">
      <h1>{this.props.props.title}</h1>
      <Icon type="close" />
    </div>
  );

  closeModal = (event) => {
    event.preventDefault();
    this.props.closeModal();
  };

  render() {
    const ChildComponent = this.props.content;
    return (
      <Modal
        width={this.props.widthSize}
        title={this.props.props.title}
        visible={this.props.isModalOpen}
        closable={false}
        footer={null}
      >
        <Popconfirm
          placement="bottomRight"
          title="Are you sure you want to cancel?"
          okText="Yes"
          cancelText="No"
          onConfirm={(event) => this.closeModal(event)}
        >
          <Button ghost className="modal__close">
            <Icon type="close" />
          </Button>
        </Popconfirm>
        <LoadingBar
          scope="modal"
          style={{
            position: "absolute",
            top: "50px",
            left: 0,
            backgroundColor: Styles.COLOR.violet,
            width: "100%",
            height: "8px",
            zIndex: 100,
          }}
        />
        {ChildComponent && (
          <ChildComponent
            closeModal={this.props.closeModal}
            clearOnSubmit={this.props.clearOnSubmit}
            {...this.props.props}
          />
        )}
      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  widthSize: getWidthSize(state),
  isModalOpen: getIsModalOpen(state),
  props: getProps(state),
  content: getContent(state),
  clearOnSubmit: getClearOnSubmit(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      closeModal,
    },
    dispatch
  );

ModalWrapper.propTypes = propTypes;
ModalWrapper.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModalWrapper);
