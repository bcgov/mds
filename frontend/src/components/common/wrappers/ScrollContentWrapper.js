import React from "react";
import { withRouter } from "react-router-dom";
import { includes } from "lodash";
import { PropTypes } from "prop-types";

/**
 * @constant ScrollContentWrapper renders react children with an active indicator if the id is in the url.
 */

const propTypes = {
  id: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  location: PropTypes.shape({ hash: PropTypes.string }).isRequired,
};

export const ScrollContentWrapper = (props) => {
  const isActive = () =>
    includes(props.id, props.location.hash) ? "circle--purple" : "circle--gray";

  return (
    <div className="scroll">
      <div className={isActive()} />
      <div>{props.children}</div>
    </div>
  );
};

ScrollContentWrapper.propTypes = propTypes;

export default withRouter(ScrollContentWrapper);
