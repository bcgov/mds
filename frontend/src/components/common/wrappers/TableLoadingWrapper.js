import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "antd";

/**
 * @constant TableLoadingWrapper renders react children or a skeleton loading view using the column headers
 * of the tables and 9 loading rows
 *
 * condition = expecting a truthy value to render children, ie "isLoaded"
 * tableHeaders = array of table headers, the skeleton loading will have the same number of columns and same titles
 * isPaginated = true if the table is paginated - this allows the skeleton to be smooth when changing pages
 *
 *
 */

const propTypes = {
  condition: PropTypes.bool.isRequired,
  children: PropTypes.element.isRequired,
  tableHeaders: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.strings)).isRequired,
  isPaginated: PropTypes.bool,
};

const defaultProps = {
  isPaginated: false,
};

export const TableLoadingWrapper = (props) => {
  const renderColumns = props.tableHeaders.map((title) => ({
    title,
    dataIndex: title,
    width: 150,
    render: () => <div className="skeleton-table__loader" />,
  }));
  return (
    <div>
      {props.condition ? (
        <div>{props.children}</div>
      ) : (
        <div className="skeleton-table">
          <Table
            align="left"
            pagination={false}
            columns={renderColumns}
            dataSource={new Array(props.isPaginated ? 25 : 9).fill({})}
            rowClassName="skeleton-table__row"
          />
        </div>
      )}
    </div>
  );
};

TableLoadingWrapper.propTypes = propTypes;
TableLoadingWrapper.defaultProps = defaultProps;

export default TableLoadingWrapper;
