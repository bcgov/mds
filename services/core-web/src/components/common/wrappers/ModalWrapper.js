import React, { Component } from "react";
import { bindActionCreators } from "redux";
import LoadingBar from "react-redux-loading-bar";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Modal, Icon, Button, Popconfirm } from "antd";
import { closeModal } from "@common/actions/modalActions";
import {
  getIsModalOpen,
  getProps,
  getContent,
  getClearOnSubmit,
  getWidth,
  getIsViewOnly,
} from "@common/selectors/modalSelectors";
import * as Styles from "@/constants/styles";
import AddPartyComponentWrapper from "./AddPartyComponentWrapper";

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool.isRequired,
  content: PropTypes.func,
  props: PropTypes.objectOf(PropTypes.any),
  clearOnSubmit: PropTypes.bool.isRequired,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  isViewOnly: PropTypes.bool.isRequired,
};

const defaultProps = {
  props: {
    title: "",
    onSubmit: () => {},
  },
  content: () => {},
};

export class ModalWrapper extends Component {
  constructor(props) {
    super(props);
    // listens for browser back || forward button click and invokes function to close the modal,
    window.onpopstate = this.onBrowserButtonEvent;
  }

  onBrowserButtonEvent = () => {
    this.props.closeModal();
  };

  closeModal = (event) => {
    event.preventDefault();
    this.props.closeModal();
    this.props.props.afterClose();
  };

  render() {
    return (
      <Modal
        width={this.props.width}
        title={this.props.props.title}
        visible={this.props.isModalOpen}
        closable={false}
        footer={null}
      >
        {this.props.isViewOnly ? (
          <Button ghost className="modal__close" onClick={(event) => this.closeModal(event)}>
            <Icon type="close" />
          </Button>
        ) : (
          <Popconfirm
            placement="bottomRight"
            title="Are you sure you want to cancel?"
            okText="Yes"
            cancelText="No"
            onConfirm={(event) => this.closeModal(event)}
          >
            <Button ghost className="modal__close">
              <Icon type="close" />
            </Button>
          </Popconfirm>
        )}
        <LoadingBar
          scope="modal"
          style={{
            position: "absolute",
            top: "50px",
            left: 0,
            backgroundColor: Styles.COLOR.violet,
            width: "100%",
            height: "8px",
            zIndex: 100,
          }}
        />
        {this.props.content && (
          <AddPartyComponentWrapper
            closeModal={this.props.closeModal}
            clearOnSubmit={this.props.clearOnSubmit}
            content={this.props.content}
            childProps={this.props.props}
          />
        )}
      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  width: getWidth(state),
  isModalOpen: getIsModalOpen(state),
  props: getProps(state),
  content: getContent(state),
  clearOnSubmit: getClearOnSubmit(state),
  isViewOnly: getIsViewOnly(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      closeModal,
    },
    dispatch
  );

ModalWrapper.propTypes = propTypes;
ModalWrapper.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModalWrapper);
