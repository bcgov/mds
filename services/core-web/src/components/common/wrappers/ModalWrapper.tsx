import React, { FC } from "react";
import { useSelector } from "react-redux";
import CommonModalWrapper from "@mds/common/wrappers/CommonModalWrapper";
import { getContent, getProps } from "@mds/common/redux/selectors/modalSelectors";
import AddPartyComponentWrapper from "./AddPartyComponentWrapper";

const ModalWrapper: FC = () => {
  const content = useSelector(getContent);
  const childProps = useSelector(getProps);

  return (
    <CommonModalWrapper>
      {content && <AddPartyComponentWrapper content={content} childProps={childProps} />}
    </CommonModalWrapper>
  );
};

export default ModalWrapper;
