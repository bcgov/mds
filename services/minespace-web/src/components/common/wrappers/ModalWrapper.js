import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Modal } from "antd";
import LoadingBar from "react-redux-loading-bar";
import { closeModal } from "@common/actions/modalActions";
import {
  getIsModalOpen,
  getProps,
  getContent,
  getClearOnSubmit,
  getWidth,
} from "@common/selectors/modalSelectors";

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool.isRequired,
  clearOnSubmit: PropTypes.bool.isRequired,
  content: PropTypes.func,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  props: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  content: () => {},
  props: {
    title: "",
    onSubmit: () => {},
  },
};

export class ModalWrapper extends Component {
  constructor(props) {
    super(props);
    // Closes modal on browser forward/back actions
    window.onpopstate = this.onBrowserButtonEvent;
  }

  onBrowserButtonEvent = () => {
    this.props.closeModal();
  };

  render() {
    const ChildComponent = this.props.content;
    return (
      <Modal
        title={this.props.props.title}
        visible={this.props.isModalOpen}
        width={this.props.width}
        footer={null}
        closable={false}
      >
        <LoadingBar
          scope="modal"
          className="color-primary"
          style={{
            position: "absolute",
            top: "50px",
            left: 0,
            width: "100%",
            height: "8px",
            zIndex: 1001,
          }}
        />
        {ChildComponent && (
          <ChildComponent
            closeModal={this.props.closeModal}
            clearOnSubmit={this.props.clearOnSubmit}
            {...this.props.props}
          />
        )}
      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  isModalOpen: getIsModalOpen(state),
  props: getProps(state),
  content: getContent(state),
  clearOnSubmit: getClearOnSubmit(state),
  width: getWidth(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      closeModal,
    },
    dispatch
  );

ModalWrapper.propTypes = propTypes;
ModalWrapper.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(ModalWrapper);
