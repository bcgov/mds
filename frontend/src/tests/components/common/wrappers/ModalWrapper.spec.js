import React from "react";
import { shallow } from "enzyme";
import { ModalWrapper } from "@/components/common/wrappers/ModalWrapper";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.closeModal = jest.fn();
  dispatchProps.content = jest.fn();
};

const setupProps = () => {
  props.props = {
    title: "mockTitle",
  };
  props.isModalOpen = false;
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("ModalWrapper", () => {
  it("renders properly", () => {
    const wrapper = shallow(<ModalWrapper {...dispatchProps} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
