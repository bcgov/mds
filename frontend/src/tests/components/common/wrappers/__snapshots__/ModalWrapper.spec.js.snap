// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalWrapper
    clearOnSubmit={false}
    closeModal={[MockFunction]}
    content={[MockFunction]}
    isModalOpen={false}
    props={
      Object {
        "title": "mockTitle",
      }
    }
    widthSize={520}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "cancelButtonDisabled": false,
      "children": Array [
        <Popconfirm
          cancelText="No"
          icon={
            <Icon
              theme="filled"
              type="exclamation-circle"
            />
          }
          okText="Yes"
          okType="primary"
          onConfirm={[Function]}
          placement="bottomRight"
          prefixCls="ant-popover"
          title="Are you sure you want to cancel?"
          transitionName="zoom-big"
          trigger="click"
        >
          <Button
            block={false}
            className="modal__close"
            ghost={true}
            loading={false}
            prefixCls="ant-btn"
          >
            <Icon
              type="close"
            />
          </Button>
        </Popconfirm>,
        <Connect(LoadingBar)
          scope="modal"
          style={
            Object {
              "backgroundColor": "#5e46a1",
              "height": "8px",
              "left": 0,
              "position": "absolute",
              "top": "50px",
              "width": "100%",
              "zIndex": 100,
            }
          }
        />,
        <mockConstructor
          clearOnSubmit={false}
          closeModal={[MockFunction]}
          title="mockTitle"
        />,
      ],
      "closable": false,
      "confirmLoading": false,
      "footer": null,
      "maskTransitionName": "fade",
      "okButtonDisabled": false,
      "okType": "primary",
      "prefixCls": "ant-modal",
      "title": "mockTitle",
      "transitionName": "zoom",
      "visible": false,
      "width": 520,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cancelText": "No",
          "children": <Button
            block={false}
            className="modal__close"
            ghost={true}
            loading={false}
            prefixCls="ant-btn"
          >
            <Icon
              type="close"
            />
          </Button>,
          "icon": <Icon
            theme="filled"
            type="exclamation-circle"
          />,
          "okText": "Yes",
          "okType": "primary",
          "onConfirm": [Function],
          "placement": "bottomRight",
          "prefixCls": "ant-popover",
          "title": "Are you sure you want to cancel?",
          "transitionName": "zoom-big",
          "trigger": "click",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": <Icon
              type="close"
            />,
            "className": "modal__close",
            "ghost": true,
            "loading": false,
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "close",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "scope": "modal",
          "style": Object {
            "backgroundColor": "#5e46a1",
            "height": "8px",
            "left": 0,
            "position": "absolute",
            "top": "50px",
            "width": "100%",
            "zIndex": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clearOnSubmit": false,
          "closeModal": [MockFunction],
          "title": "mockTitle",
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cancelButtonDisabled": false,
        "children": Array [
          <Popconfirm
            cancelText="No"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            okText="Yes"
            okType="primary"
            onConfirm={[Function]}
            placement="bottomRight"
            prefixCls="ant-popover"
            title="Are you sure you want to cancel?"
            transitionName="zoom-big"
            trigger="click"
          >
            <Button
              block={false}
              className="modal__close"
              ghost={true}
              loading={false}
              prefixCls="ant-btn"
            >
              <Icon
                type="close"
              />
            </Button>
          </Popconfirm>,
          <Connect(LoadingBar)
            scope="modal"
            style={
              Object {
                "backgroundColor": "#5e46a1",
                "height": "8px",
                "left": 0,
                "position": "absolute",
                "top": "50px",
                "width": "100%",
                "zIndex": 100,
              }
            }
          />,
          <mockConstructor
            clearOnSubmit={false}
            closeModal={[MockFunction]}
            title="mockTitle"
          />,
        ],
        "closable": false,
        "confirmLoading": false,
        "footer": null,
        "maskTransitionName": "fade",
        "okButtonDisabled": false,
        "okType": "primary",
        "prefixCls": "ant-modal",
        "title": "mockTitle",
        "transitionName": "zoom",
        "visible": false,
        "width": 520,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cancelText": "No",
            "children": <Button
              block={false}
              className="modal__close"
              ghost={true}
              loading={false}
              prefixCls="ant-btn"
            >
              <Icon
                type="close"
              />
            </Button>,
            "icon": <Icon
              theme="filled"
              type="exclamation-circle"
            />,
            "okText": "Yes",
            "okType": "primary",
            "onConfirm": [Function],
            "placement": "bottomRight",
            "prefixCls": "ant-popover",
            "title": "Are you sure you want to cancel?",
            "transitionName": "zoom-big",
            "trigger": "click",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": <Icon
                type="close"
              />,
              "className": "modal__close",
              "ghost": true,
              "loading": false,
              "prefixCls": "ant-btn",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "close",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "scope": "modal",
            "style": Object {
              "backgroundColor": "#5e46a1",
              "height": "8px",
              "left": 0,
              "position": "absolute",
              "top": "50px",
              "width": "100%",
              "zIndex": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clearOnSubmit": false,
            "closeModal": [MockFunction],
            "title": "mockTitle",
          },
          "ref": null,
          "rendered": null,
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
